internal/stream/stream.go

```diff
@@ -134,6 +165,31 @@ func (ps *PushStream) Stop() error {
 	return nil
 }
 
+// Restart is used to restart the push subscription processing
+func (ps *PushStream) Restart(ctx context.Context) {
+	logger.Ctx(ps.ctx).Infow("worker: push stream restart invoked", "subscription", ps.subscription.Name)
+	err := ps.Stop()
+	if err != nil {
+		logger.Ctx(ctx).Errorw(
+			"worker: push stream stop error",
+			"subscription", ps.subscription.Name,
+			"error", err,
+		)
+		return
+	}
+	go func(ctx context.Context) {
```

Comment:
Do not run goroutines that do not exit on ctx.Done() and other signals. these have the potential to end up being zombie routines.

Reply:
Start is a waiting call.. If ctx is done, then start will exit, and so will this go routine